stdlib/http/src/main/ballerina/src/http/cookie/cookie.bal

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'int as ints;
+import ballerina/stringutils;
+import ballerina/time;
+
+# Represents a Cookie.
+# 
+# + name - Name of the cookie
+# + value - Value of the cookie
+# + path - URI path to which the cookie belongs
+# + domain - Host to which the cookie will be sent
+# + maxAge - Maximum lifetime of the cookie represented as the number of seconds until the cookie expires
+# + expires - Maximum lifetime of the cookie represented as the date and time at which the cookie expires
+# + httpOnly - Cookie is sent only to HTTP requests
+# + secure - Cookie is sent only to secure channels
+# + creationTime - Creation time of the cookie
+# + lastAccessedTime - Last-accessed time of the cookie
+# + hostOnly - Cookie is sent only to the requested host
+public type Cookie object {
+
+    public string name = "";
```

Comment:
Shouldn't this object have an `__init()` method instead of setting empty defaults?

That way we can avoid the possibly inaccurate defaults and we can pass values as arguments to the init function instead of setting them later like at https://github.com/ballerina-platform/ballerina-lang/pull/20036/files#diff-2634164d6c4a50b656983d4f408af9daR197